server/data/data-api/pom.xml

```diff
@@ -39,25 +39,18 @@
         </dependency>
         <dependency>
             <groupId>${james.groupId}</groupId>
-            <artifactId>james-server-mailrepository-api</artifactId>
+            <artifactId>testing-base</artifactId>
+            <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>${james.groupId}</groupId>
-            <artifactId>testing-base</artifactId>
+            <groupId>io.projectreactor</groupId>
```

Comment:
Aren't these interfaces already part of Java 9? 
https://www.reactive-streams.org/ . 

We could drop the dependency? 
I just skimmed over the docs, but that is my quick assessment.

Reply:
Reactive streams might be but reactor definitly is not.